fix HNL decays

cecilecaillol opened the pull request:
Allow for intermediate Z and H in HNL decay

sihyunjeon commented:
Hello @cecilecaillol ,
I have a lot to say about this as I was digging into this spinmode onshell back in the past for analysis I was participating in (EXO-17-028). But at the end, our conclusion after discussing with Andreas Albert and Marco Trovato (EXO MC&I conveners 2 years ago) and Olivier Mattelaer (MG author) was that this mode is not fully trustable yet [1].

And actually, if your signals are targeting for mn1 > mw,mz,mh (=> mn1 > 125 GeV) you don't have to use spinmode onshell.
Adding the lines below to your MadSpin card will do the trick.
decay n1 > v z, z > ldecay ldecay
decay n1 > v h, h > ldecay ldecay

spinmode onshell was developed to treat 3 body decays with spin correlation kept alive where you really have to FORCE the particles to 3 body decay. Here, you can sequentially decay the particle n1 to bosons and bosons to leptons/neutrinos so don't really have to use spinmode onshell (spinmode default will keep the spin correlations for 2 body decays).
If you want 3 body decay, write MadSpin card more neatly in a single line, "decay n1 > ldecay ldecay v" WITH spinmode none will work just fine but your leptons and neutrinos will lose spin correlations they have.

But for the moment, I suggest you to either 1) write the sequential decay in MadSpin with spinmode default OR 2) write the 3 body decay in MadSpin with spinmode none.

I will try to get this spinmode onshell functional, and see if the problem I saw in [1] still persists as one of the items maybe next year. Thanks for reminding me of this, didn't have time to look deeper into it. If you insist on getting this into your signal samples faster skype me so that we can discuss.

[1] https://answers.launchpad.net/mg5amcnlo/+question/662694

cecilecaillol commented:
Thank you! I will try the sequential decay and let you know. What do you recommend for mn1 < mw?

cecilecaillol commented:
I have tried:

set spinmode default
decay n1 > l w, w > ldecay v
decay n1 > v z, z > ldecay ldecay
decay n1 > v h, h > ldecay ldecay

But I get this error:
spinmode can only take one of those 3 value: full/onshell/none

Which spinmode should I use with these sequential decays?

sihyunjeon commented:
For low masses, you can either use spinmode none or spinmode onshell. But as I said earlier, onshell isn't really trustable at least to me and as far as I know nobody in CMS looked deeper into this so I don't think it's good enough to be used. So although you lose spin correlations, I'd suggest you to do below.
spinmode none
decay n1 > lep lep nu

For high masses, take a look at [1] as well. Anyhow, doing like below will be my suggestion.
spinmode madspin #madspin is the name of default spinmode in MGv26X
decay n1 > w lep, w > lep nu
decay n1 > z nu, z > lep lep
decay n1 > h nu, h > lep lep

[1] is one of the hypernews that explains "spinmode none + decay n1 > lep lep nu" doesn't work for high masses since the decay table doesn't have "n1 > lep lep nu" written in it.

[1] https://hypernews.cern.ch/HyperNews/CMS/get/generators/4769.html

sihyunjeon commented:
> I have tried:
> 
> set spinmode default
> decay n1 > l w, w > ldecay v
> decay n1 > v z, z > ldecay ldecay
> decay n1 > v h, h > ldecay ldecay
> 
> But I get this error:
> spinmode can only take one of those 3 value: full/onshell/none
> 
> Which spinmode should I use with these sequential decays?

The name of spinmode changed in the middle of MGv26X or maybe MGv27X I don't remember. But if you just don't write anything for the spinmode, it would be set to default which you are trying to use for high masses.

cecilecaillol commented:
I have made a new directory with cards valid for the different mass ranges (mN < mW, mW < mN < mZ, mN> mZ). The case with high mass has explicit decays to W and Z bosons and uses spinmode full; I didnt add H decays because they are not supported in the model. In the low mass case I do 3-body decays n1 > llnu, but this works only with spinmode none.

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_M-1000_V-0.01_tau_NLO/HeavyNeutrino_trilepton_M-1000_V-0.01_tau_NLO_madspin_card.dat:
```diff
@@ -1,8 +1,8 @@
 set Nevents_for_max_weigth 250
 set max_weight_ps_point 400
-set spinmode madspin
+set spinmode onshell
```
spinmode onshell for mN=1000 GeV?

sihyunjeon replied:
Here you can just split the decays (into Z or W) with spinmode full.
Do you intend to use this?

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNgtmWltmZ_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_proc_card.dat:
```diff
@@ -0,0 +1,25 @@
+import model SM_HeavyN_NLO
+define p = p b b~
+define j = p
+generate p p > n1 l [QCD]
```
Can you change this to "generate p p > l n1 [QCD]" with reasons in [1]?

[1] https://answers.launchpad.net/mg5amcnlo/+question/661535

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNgtmWltmZ_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_run_card.dat:
```diff
@@ -0,0 +1,168 @@
+#***********************************************************************
+  1000	=  nevents  ! Number of unweighted events requested 
+  0.001	=  req_acc  ! Required accuracy (-1=auto determined from nevents)
+  100	=  nevt_job ! Max number of events per job in event generation. 
```
please change this to -1

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNgtmWltmZ_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_run_card.dat:
```diff
@@ -0,0 +1,168 @@
+# PDF choice: this automatically fixes also alpha_s(MZ) and its evol.  *
+#***********************************************************************
+  lhapdf	=  pdlabel    ! PDF set
+  260000	=  lhaid      ! if pdlabel=lhapdf, this is the lhapdf number
```
Is there any reason to choose this lhapdf set?

cecilecaillol replied:
Isnt it the recommended 5-flavor pdf in 2016?

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNgtmWltmZ_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_run_card.dat:
```diff
@@ -0,0 +1,168 @@
+  2.0	=  rw_Fscale_up      ! upper bound for fact scale variations
+  True	=  reweight_PDF      ! reweight to get PDF uncertainty
+  292201	=  PDF_set_min       ! First of the error PDF sets 
+  292302	=  PDF_set_max       ! Last of the error PDF sets
```
Remove PDF_set_min and PDF_set_max

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNgtmZ_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_proc_card.dat:
```diff
@@ -0,0 +1,25 @@
+import model SM_HeavyN_NLO
+define p = p b b~
+define j = p
+generate p p > n1 l [QCD]
```
generate p p > l n1 [QCD]

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNgtmZ_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_proc_card.dat:
```diff
@@ -0,0 +1,25 @@
+define v = vl vl~
+define w = w- w+
+import model SM_HeavyN_NLO
+define p = p b b~
```
Is this model 4f or 5f?

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNgtmZ_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_run_card.dat:
```diff
@@ -0,0 +1,168 @@
+#***********************************************************************
+  1000	=  nevents  ! Number of unweighted events requested 
+  0.001	=  req_acc  ! Required accuracy (-1=auto determined from nevents)
+  100	=  nevt_job ! Max number of events per job in event generation. 
```
-1

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNgtmZ_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_run_card.dat:
```diff
@@ -0,0 +1,168 @@
+# PDF choice: this automatically fixes also alpha_s(MZ) and its evol.  *
+#***********************************************************************
+  lhapdf	=  pdlabel    ! PDF set
+  260000	=  lhaid      ! if pdlabel=lhapdf, this is the lhapdf number
```
why not NNPDF31?

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNgtmZ_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_run_card.dat:
```diff
@@ -0,0 +1,168 @@
+  2.0	=  rw_Fscale_up      ! upper bound for fact scale variations
+  True	=  reweight_PDF      ! reweight to get PDF uncertainty
+  292201	=  PDF_set_min       ! First of the error PDF sets 
+  292302	=  PDF_set_max       ! Last of the error PDF sets
```
remove PDF_set_min PDF_set_max

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_madspin_card.dat:
```diff
@@ -0,0 +1,7 @@
+set ms_dir ./madspingrid
+set Nevents_for_max_weigth 250
+set max_weight_ps_point 400
+set spinmode full
```
Shouldn't this be spinmode none?

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_proc_card.dat:
```diff
@@ -0,0 +1,25 @@
+define v = vl vl~
+define w = w- w+
+import model SM_HeavyN_NLO
+define p = p b b~
```
4f? 5f?

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_proc_card.dat:
```diff
@@ -0,0 +1,25 @@
+import model SM_HeavyN_NLO
+define p = p b b~
+define j = p
+generate p p > n1 l [QCD]
```
generate p p > l n1 [QCD]

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_run_card.dat:
```diff
@@ -0,0 +1,168 @@
+#***********************************************************************
+  1000	=  nevents  ! Number of unweighted events requested 
+  0.001	=  req_acc  ! Required accuracy (-1=auto determined from nevents)
+  100	=  nevt_job ! Max number of events per job in event generation. 
```
-1

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0.01_tau_NLO/HeavyNeutrino_trilepton_V-0.01_tau_NLO_run_card.dat:
```diff
@@ -0,0 +1,168 @@
+# PDF choice: this automatically fixes also alpha_s(MZ) and its evol.  *
+#***********************************************************************
+  lhapdf	=  pdlabel    ! PDF set
+  260000	=  lhaid      ! if pdlabel=lhapdf, this is the lhapdf number
```
why not NNPDF31?

sihyunjeon reviewed (commented):
Hi, I tried to look at your cards but there are actually too many lol.

But I have some comments in general
nevt_job should be -1
Reason for using 260000 pdfset
Removing PDF_set_min and PDF_set_max

sihyunjeon commented:
So let's converge the idea of madspin usage through comments and then make the pull request later.
1. mN < mW : spinmode none + n1 > ldecay ldecay v
2. mW < mN < mZ : spinmode full + n1 > ldecay w, w > ldecay v
3. mN> mZ : spinmode full + n1 > ldecay w, w > ldecay v + n1 > v z, z > ldecay ldecay

Is this correct?

cecilecaillol commented:
Thanks!

    1. mN < mW : spinmode none + n1 > ldecay ldecay v -> yes, I will change to none, the others are not working
    2. mW < mN < mZ : spinmode full + n1 > ldecay w, w > ldecay v -> I thought onshell was equivalent, I will try to use full
    3. mN> mZ : spinmode full + n1 > ldecay w, w > ldecay v + n1 > v z, z > ldecay ldecay -> same as 2.

cecilecaillol commented:
Reason for using 260000 pdfset -> I thought this was the recommended 5-flavour pdf in 2016. Can you let me know what the 4-f and 5-f recommendations are?

sihyunjeon commented:
Also I see you make signal samples with mN=80,82,92. Here narrow width approximation is not really a good approach since the mass differences are too small. Did you check your sample's cross sections and the branching ratios have no issue?

sihyunjeon commented:
> Reason for using 260000 pdfset -> I thought this was the recommended 5-flavour pdf in 2016. Can you let me know what the 4-f and 5-f recommendations are?

Are you making the gridpacks for UL?
$DEFAULT_PDF_SETS = lhaid
writing the line like above will store all pdfweights in [1]

[1] https://github.com/cms-sw/genproductions/blob/master/MetaData/pdflist_5f_2017.dat

cecilecaillol commented:
We are not planning to use UL and I thought in that case for 2016 we need to write explicitly a different PDF number?

sihyunjeon commented:
@cecilecaillol Post Moriond analysis should switch to ultra legacy and this is a general guide line. Is there any special reason?

cecilecaillol commented:
We are hoping to be ready soon after Moriond, and this will be hard if we need to switch to UL as most samples, event bkg, are not available yet. We have raised this point but have not received an explicit greenlight from conveners/PC.

cecilecaillol commented:
I think I have applied all your comments. Does this look good now?

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/createCards_mNgtmWltmZ.sh:
```diff
@@ -0,0 +1,24 @@
+masspoints=(90)
```
Can you check whether 90 GeV mass point has reasonable cross section and decay BRs? NWA won't be applicable

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/HeavyNeutrino_trilepton_V-0.01_tau_NLO/createCards_mNltmW.sh:
```diff
@@ -0,0 +1,24 @@
+masspoints=(20 30 40 50 60 70 80)
```
80 is also unstable with NWA

sihyunjeon reviewed (commented):
Can you also change 0.01 to 0p01?
I sometimes saw errors when there was "." in the middle of the names.

sihyunjeon commented:
> We are not planning to use UL and I thought in that case for 2016 we need to write explicitly a different PDF number? In that case would that be 292600 for the 2016 gridpacks only?

@agrohsje might be able to help you better on this

cecilecaillol commented:
I have changed 0.01 to 0p01 and removed the masses 80 and 90 GeV.

sihyunjeon commented:
@cecilecaillol Can you only upload 1 representative mass point and the card generating script? It's hard to review them all since it's too long :(

cecilecaillol commented:
I have uploaded only the template for all mass points and the scripts for the LO case, for both the 2016 and 2017+2018 settings. You can look at https://github.com/cms-sw/genproductions/tree/d1dbd958e5b767c63ff54d54fd55e2eb5a5e398e/bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0p01_tau_LO and https://github.com/cms-sw/genproductions/blob/d1dbd958e5b767c63ff54d54fd55e2eb5a5e398e/bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/createCards_mNltmW_LO.sh
Thanks!

sihyunjeon commented:
Hi Cecile,

Can you send me a skype message?

Regards,
Sihyun Jeon.

L3 Convener of CMS-EXO MC & Interpretations Subgroup.
PhD Student of Seoul National University,
Department of Physics & Astronomy, Seoul National University,
1 Gwanak-ro, Gwanak-gu, Seoul, Korea,
Bldg. 23, Room. 305.

Skype : sihyun_jeon
Email : shjeon@cern.ch
Phone : (+82)10-6357-6284

On Jan 5, 2021 10:21 PM +0900, cecilecaillol <notifications@github.com>, wrote:
> I have uploaded only the template for all mass points and the scripts for the LO case, for both the 2016 and 2017+2018 settings. You can look at https://github.com/cms-sw/genproductions/tree/d1dbd958e5b767c63ff54d54fd55e2eb5a5e398e/bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0p01_tau_LO and https://github.com/cms-sw/genproductions/blob/d1dbd958e5b767c63ff54d54fd55e2eb5a5e398e/bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/createCards_mNltmW_LO.sh
> Thanks!
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0p01_tau_LO/HeavyNeutrino_trilepton_V-0p01_tau_LO_extramodels.dat:
```diff
@@ -0,0 +1,2 @@
+SM_HeavyN_AllMassive_LO.tgz
+SM_HeavyN_Dirac_CKM_Masses_LO_UFO.tgz
```
Why two extramodels?

cecilecaillol replied:
I have removed the Dirac model.

sihyunjeon commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0p01_tau_NLO/HeavyNeutrino_trilepton_V-0p01_tau_NLO_madspin_card.dat:
```diff
@@ -0,0 +1,7 @@
+set spinmode none
+set max_running_process 1
+decay n1 > v ldecay ldecay
+launch
```
Should we remove this mN<mW cards for NLO? As you are going to use LO for these mass points?

cecilecaillol replied:
I will remove them, we are now planning to use LO.

cecilecaillol commented:
The cards should be final now. I have cleaned the individual mass points, aligned the LO run_card with your recent example, updated the 2016 PDFs, and kept only the Majorana neutrino model at LO (same as for the NLO samples at high mass).

sihyunjeon commented:
Okay, the cards are good for me.

@agrohsje the cards are a bit divergent. So this might confuse you
1) mN < mW : LO process (for spin correlation considerations)
2) mW < mN < mZ : NLO process with madspin only decaying N to W
3) mN > mZ : NLO process with madspin decaying N to W and Z
And they have two set of cards for 2016 (pdf specified) and 2017/2018 (defaultpdf settings)

So 6 set of cards in total. @cecilecaillol correct me if I am wrong

agrohsje commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/HeavyNeutrino_trilepton_mNgtmWltmZ_V-0p01_tau_NLO/HeavyNeutrino_trilepton_V-0p01_tau_NLO_customizecards.dat:
```diff
@@ -1,4 +1,4 @@
-set param_card mass 9900012 1000
+set param_card mass 9900012 HNLMASS
```
What is the width in this case? Is the model default ok?

cecilecaillol replied:
Thank you for the checks! In the last line of the same file the width is recomputed for each mass point so it should be ok.

agrohsje replied:
I see. Is only displayed when clicking on view file. Ok.

agrohsje commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/HeavyNeutrino_trilepton_mNgtmZ_V-0p01_tau_NLO/HeavyNeutrino_trilepton_V-0p01_tau_NLO_customizecards.dat:
```diff
@@ -1,4 +1,4 @@
-set param_card mass 9900012 1000
+set param_card mass 9900012 HNLMASS
```
as above. what about the width ?

cecilecaillol replied:
Same, we recompute the width in customizecards

agrohsje commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/HeavyNeutrino_trilepton_mNgtmZ_V-0p01_tau_NLO/HeavyNeutrino_trilepton_V-0p01_tau_NLO_run_card.dat:
```diff
@@ -58,10 +58,8 @@
-  $DEFAULT_PDF_MEMBERS  = reweight_PDF
-   
+  lhapdf	=  pdlabel    ! PDF set
+  292200	=  lhaid      ! if pdlabel=lhapdf, this is the lhapdf number
```
How do you distinguish 2016 from 2017/2018 here. I don't see this indicated from the path name.

cecilecaillol replied:
We have different sets of cards and commits for 2016 and for 2017/2018. Sorry if this commit is confusing. In 2017/2018 we take the DEFAULT_PDF_MEMBERS and in 2016 we specify the lhaid

agrohsje commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0p01_tau_LO/HeavyNeutrino_trilepton_V-0p01_tau_LO_proc_card.dat:
```diff
@@ -0,0 +1,24 @@
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_optimized_output True
+set loop_color_flows False
+set gauge unitary
+set complex_mass_scheme False
+set max_npoint_for_channel 0
+import SM_HeavyN_AllMassive_LO
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
```
Please remove these default lines. Is the model 5FS or 4FS per default?

agrohsje commented on bin/MadGraph5_aMCatNLO/cards/production/13TeV/HeavyNeutrino_trilepton_V-0p01_tau_NLO/HeavyNeutrino_trilepton_mNltmW_V-0p01_tau_LO/HeavyNeutrino_trilepton_V-0p01_tau_LO_run_card.dat:
```diff
@@ -0,0 +1,259 @@
+# PDF CHOICE: this automatically fixes also alpha_s and its evol.    *
+#*********************************************************************
+  lhapdf        =  pdlabel    ! PDF set
+  262000        =  lhaid      ! if pdlabel=lhapdf, this is the lhapdf number
```
As above. Make sure variations are in runcmsgrid.sh.

cecilecaillol replied:
Thanks to Sihyun I confirm the variations are in runcmsgrid.sh

sihyunjeon commented:
Ahhh... I forgot the width.
With this model, the V value (set to 0.01 here) will control the width. And even with default width, it is possible to set the limit results but it should add a bit more work. But should be easier to consider it during the gridpack production... I'll ping the requestors.

sihyunjeon commented:
@agrohsje
Sorry maybe we missed some lines. The width is computed here [1]

[1] https://github.com/cms-sw/genproductions/pull/2790/files#diff-94cc8529320a925a677da80af5652b8ef13159d1657026bae88b48bee6c4562fR11

agrohsje commented:
Let me merge the PR.
